_posts/2020/2020-01-20-be-suspicious-of-join-tables.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Be Suspicious of Join Tables"
+description: "They're probably hiding a concept you haven’t named yet"
+layout: article
+category: ruby
+image:
+  base: '2020/be-suspicious-of-join-tables'
+  alt: Welding
+  credit:  Max LaRochelle
+  source: "https://unsplash.com/photos/QzP1GcDOSC8"
+
+---
+
+We often have to represent many-to-many relationships between models in our applications. Rails provides a method in the migrations to generate a table in your database to support this. You can see the documentation in the [Rails guide for ActiveRecord migrations](https://guides.rubyonrails.org/active_record_migrations.html#creating-a-join-table).
+
+However these “pure” join tables often obscure a useful concept in your application that might be better represented as a named model.
+
+
+## Instead of...
+
+...using a join table:
+
+#### Migration
+
+```ruby
+create_join_table :user, :organisation
+```
+
+#### `app/models/user.rb`
+
+```ruby
+class User < ApplicationRecord
+  has_and_belongs_to_many :organisations
+end
+```
+
+#### `app/models/organisation.rb`
+
+```ruby
+class Organisation < ApplicationRecord
+  has_and_belongs_to_many :users
+end
+```
+
+## Use
+
+...a real model, by naming the concept that the join table is hiding.
```

Comment:
*...a real model _and_ (?) name the concept that the join table is hiding.

Reply:
Because you could technically name your real model the same as what its autogenerated by the create_join_table method.